src/riak_kv_wm_timeseries.erl

```diff
@@ -29,572 +34,375 @@
-call_api_function(RD, Ctx = #ctx{api_call = put,
-                                 table = Table, data = Data}) ->
-    Mod = riak_ql_ddl:make_module_name(Table),
-    %% convert records to tuples, just for put
-    Records = [list_to_tuple(R) || R <- Data],
-    case riak_kv_ts_util:validate_rows(Mod, Records) of
-        [] ->
-            case riak_kv_ts_api:put_data(Records, Table, Mod) of
-                ok ->
-                    prepare_data_in_body(RD, Ctx#ctx{result = ok});
-                {error, {some_failed, ErrorCount}} ->
-                    handle_error({failed_some_puts, ErrorCount, Table}, RD, Ctx);
-                {error, no_ctype} ->
-                    handle_error({no_such_table, Table}, RD, Ctx)
-            end;
-        BadRowIdxs when is_list(BadRowIdxs) ->
-            handle_error({invalid_data, BadRowIdxs}, RD, Ctx)
-    end;
-
-call_api_function(RD, Ctx0 = #ctx{api_call = get,
-                                  table = Table, key = Key, mod = Mod,
-                                  timeout = Timeout}) ->
-    Options =
-        if Timeout == undefined -> [];
-           true -> [{timeout, Timeout}]
-        end,
-    case riak_kv_ts_api:get_data(Key, Table, Mod, Options) of
-        {ok, Record} ->
-            {ColumnNames, Row} = lists:unzip(Record),
-            %% ColumnTypes = riak_kv_ts_util:get_column_types(ColumnNames, Mod),
-            %% We don't need column types here as well (for the PB interface, we
-            %% needed them in order to properly construct tscells)
-            DataOut = {ColumnNames, [Row]},
-            %% all results (from get as well as query) are returned in
-            %% a uniform 'tabular' form, hence the [] around Row
-            Ctx = Ctx0#ctx{result = DataOut},
-            prepare_data_in_body(RD, Ctx);
-        {error, notfound} ->
-            handle_error(notfound, RD, Ctx0);
-        {error, {bad_key_length, Got, Need}} ->
-            handle_error({key_element_count_mismatch, Got, Need}, RD, Ctx0);
-        {error, Reason} ->
-            handle_error({riak_error, Reason}, RD, Ctx0)
-    end;
-
-call_api_function(RD, Ctx = #ctx{api_call = delete,
-                                 table = Table, key = Key,
-                                 mod = Mod,
-                                 timeout = Timeout}) ->
-    Options =
-        if Timeout == undefined -> [];
-           true -> [{timeout, Timeout}]
-        end,
-    case riak_kv_ts_api:delete_data(Key, Table, Mod, Options) of
-        ok ->
-            prepare_data_in_body(RD, Ctx#ctx{result = ok});
-        {error, {bad_key_length, Got, Need}} ->
-            handle_error({key_element_count_mismatch, Got, Need}, RD, Ctx);
-        {error, notfound} ->
-            handle_error(notfound, RD, Ctx);
-        {error, Reason} ->
-            handle_error({riak_error, Reason}, RD, Ctx)
-    end.
+%% @todo: might be better if the DDL helper module had a
+%% valid_field_value(Field, Value) -> boolean() function.
```

Comment:
There is one, called `Mod:validate_obj/1`. It operates on an entire record, though, but you can still consider using it.